source/m2mfirmware.cpp

```diff
@@ -430,5 +430,7 @@ bool M2MFirmware::check_value_range(FirmwareResource resource, int64_t value) co
 
 void M2MFirmware::set_zero_value(M2MResource *resource)
 {
-    resource->set_value((const uint8_t*)"0", 1);
+    if(resource) {
+        resource->set_value((const uint8_t*)"0", 1);
+    }
```

Comment:
You could try "assert(&res != NULL)" but IMHO there is no point to validate against malicious intent of the developer. :)